06_predict_survival/nbconverted/plot_survival_pc_comparison.py

```diff
@@ -0,0 +1,459 @@
+training_data_map = {
+    'expression': 'gene expression',
+    'me_27k': '27k methylation',
+    'me_450k': '450k methylation',
+    'rppa': 'RPPA',
+    'mirna': 'microRNA',
+    'mut_sigs': 'mutational signatures',
+}
+
+
+# In[13]:
+
+
+all_data_count_df = []
+
+for n_dim in pcs_list:
+    # load results into a single dataframe
+    all_pcs_dir = Path(all_results_dir, 'results_{}_pca'.format(n_dim))
+    all_results_df = su.load_survival_results(all_pcs_dir)
+    all_results_df.rename(columns={'identifier': 'cancer_type',
+                                  'fold_no': 'fold'}, inplace=True)
+    
+    all_count_df = (all_results_df[all_results_df.data_type == 'test']
+        .groupby(['cancer_type', 'training_data'])
+        .count()
+    )
+    problem_df = all_count_df[all_count_df['cindex'] != 16].copy()
+    drop_cancer_types = problem_df.index.get_level_values(0).unique().values
+    
+    all_all_results_df = au.compare_all_data_types(
+        all_results_df[~all_results_df.cancer_type.isin(drop_cancer_types)],
+        SIG_ALPHA,
+        identifier='cancer_type',
+        metric='cindex')
+    
+    print(n_dim)
+    num_reject = all_all_results_df.groupby(['training_data']).sum().reject_null
+    num_total = all_all_results_df.groupby(['training_data']).count().reject_null
+    
+    dim_count_df = (
+            pd.DataFrame(num_reject).rename(columns={'reject_null': 'num_reject'})
+        .merge(
+            pd.DataFrame(num_total).rename(columns={'reject_null': 'num_total'})
+        , left_index=True, right_index=True)
+    )
+    dim_count_df['n_dim'] = n_dim
+    dim_count_df['ratio'] = dim_count_df.num_reject / dim_count_df.num_total
+    all_data_count_df.append(dim_count_df)
+    
+all_data_count_df = pd.concat(all_data_count_df).reset_index()
+all_data_count_df.training_data.replace(to_replace=training_data_map, inplace=True)
+all_data_count_df.head(10)
+
+
+# In[14]:
+
+
+sns.set({'figure.figsize': (22, 5)})
+sns.set_style('whitegrid')
+fig, axarr = plt.subplots(1, 3)
+
+sns.pointplot(data=all_data_count_df, x='n_dim', y='num_reject',
+              hue='training_data', hue_order=training_data_map.values(), ax=axarr[0])
+axarr[0].set_xlabel('Number of PCs')
+axarr[0].set_ylabel('Number of well-predicted cancer types')
+axarr[0].set_title('Well-predicted cancer type count vs. PC count')
+
+sns.pointplot(data=all_data_count_df, x='n_dim', y='num_total',
+              hue='training_data', hue_order=training_data_map.values(), ax=axarr[1])
+axarr[1].set_xlabel('Number of PCs')
+axarr[1].set_ylabel('Total valid cancer types')
+axarr[1].set_title('Valid cancer type count vs. PC count')
+
+sns.pointplot(data=all_data_count_df, x='n_dim', y='ratio',
+              hue='training_data', hue_order=training_data_map.values(), ax=axarr[2])
+axarr[2].set_xlabel('Number of PCs')
+axarr[2].set_ylabel('Ratio of well-predicted to total cancer type')
+axarr[2].set_title('Well-predicted/total ratio vs. PC count')
```

Comment:
Disregard previous comment, proteome = best 'ome

Is there a reason you split out expression + methylation in their own figures, then do all data types? Are the other data types run on a subset of the samples or something?

Reply:
The two separate experiments (for expression/methylation, and all data types) is the same thing we're doing for mutation prediction as well. The main reason is that in each comparison, we're using only the TCGA samples that have data for _all_ of the included data types.

That means we have about 3,000 fewer samples when we combine all data types as opposed to just looking at expression and methylation alone. So we did two separate experiments, under the assumption that we might be better powered to see some effects in the expression/methylation comparison that don't come through in the comparison of all data types.

The overall trends are pretty much the same whether we include all data types or just expression/methylation, though, so it doesn't matter too much for the main messages of the paper.